drivers/modem/gsm_ppp.c

```diff
@@ -383,19 +389,34 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 	}
 
 	/* Don't initialize PPP until we're attached to packet service */
-	ret = modem_cmd_send_nolock(&gsm->context.iface,
-				    &gsm->context.cmd_handler,
-				    &check_attached_cmd, 1,
-				    "AT+CGATT?",
-				    &gsm->sem_response,
-				    GSM_CMD_SETUP_TIMEOUT);
+	ret = modem_cmd_send_nolock(
+		&gsm->context.iface, &gsm->context.cmd_handler,
+		&check_attached_cmd, 1, "AT+CGATT?", &gsm->sem_response,
+		GSM_CMD_SETUP_TIMEOUT);
 	if (ret < 0) {
+		/*
+		 * attach_retry not set        -> trigger 10 attach retries
+		 * attach_retry set            -> decrement and retry
+		 * attach_retry set, becomes 0 -> trigger full retry
+		 *
+		 * In total, modem gets 60 seconds to ack AT+CGATT which should
+		 * be fine for most RATs but possibly a making num retries
+		 * Kconfigurable would be best eventually.
```

Comment:
Having this configurable would be great, it should not be too big change to add that to this PR too